tests/test_discounts.py

```diff
@@ -201,6 +201,56 @@ def test_invalid_checkout_discount_form(monkeypatch, voucher):
     assert 'voucher' in form.errors
 
 
+def test_voucher_queryset_active(voucher):
+    vouchers = Voucher.objects.all()
+    assert len(vouchers) == 1
+    active_vouchers = Voucher.objects.active(
+        date=date.today() - timedelta(days=1))
+    assert len(active_vouchers) == 0
+
+
+def test_checkout_discount_form_active_queryset_voucher_not_active(voucher):
+    assert len(Voucher.objects.all()) == 1
+    checkout = Mock(cart=Mock())
+    voucher.start_date = date.today() + timedelta(days=1)
+    voucher.save()
+    form = CheckoutDiscountForm({'voucher': voucher.code}, checkout=checkout)
+    qs = form.fields['voucher'].queryset
+    assert len(qs) == 0
+
+
+def test_checkout_discount_form_active_queryset_voucher_active(voucher):
+    assert len(Voucher.objects.all()) == 1
+    checkout = Mock(cart=Mock())
+    voucher.start_date = date.today()
+    voucher.save()
+    form = CheckoutDiscountForm({'voucher': voucher.code}, checkout=checkout)
+    qs = form.fields['voucher'].queryset
+    assert len(qs) == 1
+
+
+@patch('saleor.discount.forms.date')
```

Comment:
Please use `freezegun` instead of mocking imported functions.

Reply:
Done